doc/source/development/rfc/rfc104_gdal_cli.rst

```diff
@@ -0,0 +1,1175 @@
+          "category":"Base"
+        },
+        {
+          "name":"output-layer",
+          "type":"string",
+          "description":"Output layer name",
+          "min_count":0,
+          "max_count":1,
+          "category":"Base"
+        }
+      ],
+      "output_arguments":[
+      ],
+      "input_output_arguments":[
+        {
+          "name":"output",
+          "type":"dataset",
+          "description":"Output vector dataset",
+          "min_count":1,
+          "max_count":1,
+          "category":"Base",
+          "dataset_type":[
+            "vector"
+          ],
+          "input_flags":[
+            "name",
+            "dataset"
+          ],
+          "output_flags":[
+            "dataset"
+          ]
+        }
+      ]
+    }
+
+.. _rfc104_gdal_vector_pipeline_examples:
+
+A few invocations of "gdal vector pipeline"
++++++++++++++++++++++++++++++++++++++++++++
+
+  .. code-block:: shell
+
+     # The use of the '!' as a step separator is to prevent Unix or Windows shells from
+     # trying to use other processes for the "reproject" or "write" steps.
+     # Below is a single-process pipeline.
+     $ gdal vector pipeline read poly.gpkg ! reproject --dst-crs=EPSG:4326 ! write out.parquet --overwrite
+
+     # Alternative without the "vector" and "pipeline" subcommands, and with --progress
+     $ gdal read poly.gpkg ! reproject --dst-crs=EPSG:4326 ! write out.parquet --overwrite  --progress
+
+     # Alternative using an explicit --pipeline switch, and given the quoting, we can use the '|' character
+     $ gdal vector pipeline --pipeline="read poly.gpkg | reproject --dst-crs=EPSG:4326 | write out.parquet --overwrite"
+
+     # Works also as a quoted positional argument, and without the "vector" subcommand
+     $ gdal pipeline --progress "read poly.gpkg | reproject --dst-crs=EPSG:4326 | write out.parquet --overwrite"
+
+
+Detailed usage help message of "gdal vector pipeline"
++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+  .. code-block:: shell
+
+    $ gdal vector pipeline --help
+    Usage: gdal vector pipeline [OPTIONS] <PIPELINE>
+
+    Process a vector dataset.
+
+    Positional arguments:
+
+    Common Options:
+      -h, --help    Display help message and exit
+      --json-usage  Display usage as JSON document and exit
+      --progress    Display progress bar
+
+    <PIPELINE> is of the form: read [READ-OPTIONS] ( ! <STEP-NAME> [STEP-OPTIONS] )* ! write [WRITE-OPTIONS]
+
+    Example: 'gdal vector pipeline --progress ! read in.gpkg ! \
+                   reproject --dst-crs=EPSG:32632 ! write out.gpkg --overwrite'
+
+    Potential steps are:
+
+    * read [OPTIONS] <INPUT>
+    ------------------------
+
+    Read a vector dataset.
+
+    Positional arguments:
+      -i, --input <INPUT>                                  Input vector dataset [required]
+
+    Options:
+      -l, --layer, --input-layer <INPUT-LAYER>             Input layer name(s) [may be repeated]
```

Comment:
3 different options for the same thing? short (1 char) and long is enough. I would remove `--input-layer` or `--layer`

Reply:
On the other hand, in documentation, tutorials, workshops, ETL type scripts which will be used by many people for a long time, etc. using `--input-layer` is self-describing but `-l` is not.  Maybe the middle-length option does not have the same advantage of clarity.